fortanix-sgx-tools/src/bin/ftxsgx-elf2sgxs.rs

```diff
@@ -201,6 +258,20 @@ impl<'a> LayoutInfo<'a> {
             bail!("ENCLAVE_SIZE symbol is not naturally aligned")
         }
 
+        if let (Some(EH_FRM_HDR_BASE), Some(EH_FRM_HDR_SIZE)) = (syms.EH_FRM_HDR_BASE, syms.EH_FRM_HDR_SIZE) {
+            check_size!(EH_FRM_HDR_BASE == 8);
+            check_size!(EH_FRM_HDR_SIZE == 8);
+        }
+        else if let (Some(EH_FRM_OFFSET), Some(EH_FRM_LEN), Some(EH_FRM_HDR_OFFSET), Some(EH_FRM_HDR_LEN)) = (syms.EH_FRM_OFFSET, syms.EH_FRM_LEN, syms.EH_FRM_HDR_OFFSET, syms.EH_FRM_HDR_LEN) {
+            check_size!(EH_FRM_OFFSET == 8);
+            check_size!(EH_FRM_LEN == 8);
+            check_size!(EH_FRM_HDR_OFFSET == 8);
+            check_size!(EH_FRM_HDR_LEN == 8);
+        }
+        else {
+            bail!("Missing EH Frame header symbols, application must either have (EH_FRM_HDR_BASE/EH_FRM_HDR_SIZE) or (EH_FRM_OFFSET, EH_FRM_LEN, EH_FRM_HDR_OFFSET, EH_FRM_HDR_LEN");
```

Comment:
I don't like that this message isn't more specific about what is missing, but I can't think of an elegant way of adding that specificity

Reply:
I tried to improve on it, if you have other suggestions let me know